### PR TITLE
Users/josue/improving file attachment error labels

### PR DESCRIPTION
**Pull request by josuhern**
**Description**
I saw that there was a gap on the error alert shown to the user when a file attachmen doesn't meet the requiments, it shows the error but doesn't tell which file caused the issue
image

**Solution Proposed**
I propose modifing the localization files to add the file name in the alert in order to make a reference about this specific file doesn't meet the requirements

**Acceptance criteria**
Show the file name that doesn't meet the requirements

**Test cases and evidence**
https://josue.blob.core.windows.net/tasks/indexv2.html
image
image

**Pop up**
image
image

### CODE REVIEW

**Comment by elopezanaya** [comment]
I suggest we limit the chars displayed in the label to avoid breaking the UI style with files with long names, by example  :  

instead of :  __this_is_a_very_very_very_very_very_very_very_long_name.xlsx__
 to
 __this_is_a_very...__

**Review by elopezanaya — CHANGES_REQUESTED** [review]
I suggest we limit the chars displayed in the label to avoid breaking the UI style with files with long names, by example :

instead of : this_is_a_very_very_very_very_very_very_very_long_name.xlsx
to
this_is_a_very...

**Comment by elopezanaya** [comment]
add test evidence for pop-out

**Comment by josuhern** [comment]
> I suggest we limit the chars displayed in the label to avoid breaking the UI style with files with long names, by example :
> 
> instead of : this_is_a_very_very_very_very_very_very_very_long_name.xlsx to this_is_a_very...

Done

**Review by elopezanaya — APPROVED** [review]

File: chat-widget/src/components/webchatcontainerstateful/webchatcontroller/middlewares/storemiddlewares/attachmentUploadValidatorMiddleware.ts [hunk 1]
```diff
@@ -77,6 +77,11 @@ const isImage = (contentType: string): boolean => {
     return AMSConstants.supportedImagesMimeTypes.includes(contentType);
 };
 
+const textEllipsis = (str: string, maxLength = 20): string => {
```
Comment (charliewang95) [inline chat-widget/src/components/webchatcontainerstateful/webchatcontroller/middlewares/storemiddlewares/attachmentUploadValidatorMiddleware.ts:80]: Please attach screenshot proof for this change

File: chat-widget/src/components/livechatwidget/common/defaultProps/dummyDefaultProps.ts [hunk 2]
```diff
@@ -1692,12 +1692,12 @@ export const dummyDefaultProps: ILiveChatWidgetProps = {
         },
         localizedTexts: {
             MIDDLEWARE_BANNER_FILE_NULL_ERROR: "There was an error uploading the file, please try again.",
-            MIDDLEWARE_BANNER_FILE_SIZE_WITHOUT_EXTENSION_ERROR: "File exceeds the allowed limit of {0} MB and please upload the file with an appropriate file extension.",
-            MIDDLEWARE_BANNER_FILE_SIZE_EXTENSION_ERROR: "File exceeds the allowed limit of {0} MB and {1} files are not supported.",
-            MIDDLEWARE_BANNER_FILE_WITHOUT_EXTENSION: "File upload error. Please upload the file with an appropriate file extension.",
+            MIDDLEWARE_BANNER_FILE_SIZE_WITHOUT_EXTENSION_ERROR: "File {0} exceeds the allowed limit of {1} MB and please upload the file with an appropriate file extension.",
```
Comment (charliewang95) [inline chat-widget/src/components/livechatwidget/common/defaultProps/dummyDefaultProps.ts:1695]: By adding {0} backward compatibility is broken. We need to clearly log this change in the Change Log, and also keep supporting the old format.

Comment (charliewang95) [inline chat-widget/src/components/livechatwidget/common/defaultProps/dummyDefaultProps.ts:1695]: Please update change log

File: CHANGE_LOG.md [hunk 3]
```diff
@@ -36,4 +36,5 @@ All notable changes to this project will be documented in this file.
 
 - Added individual UI components like `Header`, `Footer`, and `ChatButton`, etc. with customizable `controlProps` and `styleProps`
 - Added `BroadcastService` to communicate with package consumer layer, and for telemetry purposes
-- Exposed `encodeComponentString` and `encodeComponentString`
\ No newline at end of file
+- Exposed `encodeComponentString` and `encodeComponentString`
+- Added MIDDLEWARE_BANNER_FILE parameter `{2}` to show the user the file name when an exception occurs, also implementing standard for parameter to keep backwards compatibility
```
Comment (charliewang95) [inline CHANGE_LOG.md:40]: 1.0.0 is already released. Put the change under the "Unreleased" -> "Added" section.

File: chat-widget/src/components/livechatwidget/common/defaultProps/dummyDefaultProps.ts [hunk 4]
```diff
@@ -1697,13 +1697,17 @@ export const dummyDefaultProps: ILiveChatWidgetProps = {
             }
         },
         localizedTexts: {
+            //MIDDLEWARE_BANNER_FILE parameters:
+            //{0} = file limit size
+            //{1} = file extension
+            //{2} = File name
```
Comment (charliewang95) [inline chat-widget/src/components/livechatwidget/common/defaultProps/dummyDefaultProps.ts:1703]: "F" casing

File: chat-widget/src/components/livechatwidget/common/defaultProps/dummyDefaultProps.ts [hunk 5]
```diff
@@ -1697,13 +1697,17 @@ export const dummyDefaultProps: ILiveChatWidgetProps = {
             }
         },
         localizedTexts: {
+            //MIDDLEWARE_BANNER_FILE parameters:
```
Comment (charliewang95) [inline chat-widget/src/components/livechatwidget/common/defaultProps/dummyDefaultProps.ts:1700]: Use /* */ for multi-line comments

File: chat-widget/src/components/webchatcontainerstateful/common/defaultProps/defaultMiddlewareLocalizedTexts.ts [hunk 6]
```diff
@@ -1,13 +1,18 @@
 import { ILiveChatWidgetLocalizedTexts } from "../../../../contexts/common/ILiveChatWidgetLocalizedTexts";
 
+//MIDDLEWARE_BANNER_FILE parameters:
```
Comment (charliewang95) [inline chat-widget/src/components/webchatcontainerstateful/common/defaultProps/defaultMiddlewareLocalizedTexts.ts:3]: same here

**Review by charliewang95 — APPROVED** [review]

**Pull request merged**